### ISSUE TITLE
[dynamo] SymInt / SymInt division not supported (3 sliding-window models, 100 breaks)

### ISSUE THREAD

**Issue by penguinwu** [penguinwu/oss-model-graph-break-corpus#99]
**Repro status:** Reproduces on torch 2.9.0+cu128 (stable, verified 2026-05-09 16:14 UTC) and torch 2.12.0.dev20260407+cu128 (nightly, verified 2026-05-09 15:54 UTC).

## Original failure report

5 sliding-window / convolution-heavy models produce 100 graph breaks from a single Dynamo gap: division on two SymInt arguments isn't yet supported.

**Test command (sweep):** `python tools/run_experiment.py sweep ...` (cohort: LongformerModel, LEDModel, LEDForConditionalGeneration in eval+train modes, animesh-fullgraph 2026-04-28).

**Symptom captured:** every break's `reason` text matches `Calling <built-in method div of type object at 0x...> on only torch.SymInt arguments is not yet supported.` (the `0x...` address differs per process; the message after the address is stable). 34 occurrences per LongformerModel run, ~33 per LED variant.

### Affected Models

| Model | Modes | Breaks |
|---|---|---|
| LEDForConditionalGeneration | eval, train | 33 |
| LEDModel | eval, train | 33 |
| LongformerModel | eval, train | 34 |

<details><summary>Verification signal (original)</summary>

`{"kind": "stderr_contains", "fragment": "on only torch.SymInt arguments is not yet supported"}`
</details>

## Pattern

Triggered when shape arithmetic computes a quotient that Dynamo represents as `SymInt / SymInt`. The cited call site at `transformers/models/longformer/modeling_longformer.py:770`:

```text
chunks_count = torch.div(seq_len, window_overlap, rounding_mode="trunc") - 1
```

where `seq_len = query.size(1)` becomes a `SymInt` under dynamic shapes and `window_overlap` is a Python int that Dynamo also lifts to `SymInt` along that axis.

## Minimal reproducer (MRE)

Reduced from `modeling_longformer.py:770` (the cited call site). 15 lines, runs in ~8s:

```python repro=true
"""
MRE candidate v1 for issue #99: SymInt/SymInt graph break in LongformerModel.
Provenance: transformers/models/longformer/modeling_longformer.py:770
    chunks_count = torch.div(seq_len, window_overlap, rounding_mode="trunc") - 1
where seq_len = query.size(1) is a SymInt under dynamic shapes and
window_overlap is a Python int constant.
"""
import torch


@torch.compile(fullgraph=True, backend="eager", dynamic=True)
def f(x, window_overlap):
    seq_len = x.size(1)
    chunks_count = torch.div(seq_len, window_overlap, rounding_mode="trunc") - 1
    return x.sum() + chunks_count


x = torch.randn(1, 1024, 8)
out = f(x, 256)
print("ok", out)
```

<details><summary>Verification signal (MRE)</summary>

`{"kind": "stderr_contains", "fragment": "on only torch.SymInt arguments is not yet supported"}`
</details>

## Source

- Sweep results: `sweep_results/experiments/animesh-fullgraph-2026-04-28/explain_results.json` (rows where `name in {LongformerModel, LEDModel, LEDForConditionalGeneration}`)
- Provenance anchor: `transformers/models/longformer/modeling_longformer.py:770`
- Built off PR pytorch/pytorch#181552
- Transformers: 5.6.2

## Related

- #98 (ParametrizedConv1d recompile-limit cascade) — same 5-model cohort historically; this issue's scope is the SymInt-div sub-pattern only.

---
<sub>Body rewritten 2026-05-09 by file-issue + mre subagent dogfood. MRE deduced from real call site (not hypothesis), verified on torch 2.9 stable + torch 2.12 nightly.</sub>
